server/controllers/documents.js

```diff
@@ -0,0 +1,202 @@
+const Document = require('../models/documents');
+const Role = require('../models/roles');
+
+module.exports = {
+  // adding a new document
+  create: ((req, res) => {
+    Role.findOne({
+      role: req.decoded.role
+    }).select('_id').exec((err, id) => {
+      if (err) {
+        throw err;
+      } else {
+        const document = new Document(); // creating an instance of Document model
+        // values to be added to new instance
+        document.title = req.body.title;
+        document.content = req.body.content;
+        document.ownerId = req.decoded._id; // console.log(req.decoded);
+        document.roleId = id;
+
+        if (req.body.view) {
+          document.view = req.body.view;
+        }
+
+        // save the document and check for errors
+        document.save((error) => {
+          if (error) {
+            res.send(error);
+          }
+          res.json({
+            message: 'document created',
+            document: document
+          });
+        });
+      }
+    });
+  }),
+
+  find: ((req, res) => {
+    let limit = req.query.limit;
+    limit = parseInt(limit, 10);
+    Document.aggregate(
+        { $sort: { createdAt: -1 } }, ((err, documents) => {
+          if (err) {
+            res.send(err);
+          } else if (req.decoded.role === 'user') {
+            documents.forEach((doc) => {
+              if (doc.ownerId === req.decoded._id || doc.view === 'public') {
```

Comment:
Could these be added as part of the query sent to MongoDB?
Fetching all documents and then looping through them while checking search criteria, does not seem scalable.